polygeist/test/polygeist-opt/sycl/sycl-raise-host.mlir

```diff
@@ -770,48 +805,54 @@ llvm.func @_ZN4sycl3_V17handler6unpackEv(%arg0: !llvm.ptr)
+// CHECK:           llvm.store %[[VAL_4]], %[[VAL_26]] : f32, !llvm.ptr
+// CHECK:           %[[VAL_27:.*]] = llvm.mlir.addressof @constant_array_0 : !llvm.ptr<array<5 x f32>>
+// CHECK:           sycl.host.set_captured %[[VAL_18]][6] = %[[VAL_27]] : !llvm.ptr, !llvm.ptr<array<5 x f32>>
+// CHECK:           %[[VAL_28:.*]] = llvm.getelementptr %[[VAL_18]][0, 7] : (!llvm.ptr) -> !llvm.ptr, !llvm.struct<"class.lambda", (i16, i32, struct<"class.sycl::_V1::accessor", (ptr)>, struct<"class.sycl::_V1::vec", (array<16 x i16>)>, f32, f32, array<5 x f32>, array<5 x f32>, struct<"class.sycl::_V1::accessor", (ptr)>)>
+// CHECK:           llvm.store %[[VAL_12]], %[[VAL_28]] : vector<2xf32>, !llvm.ptr
+// CHECK:           %[[VAL_29:.*]] = llvm.mlir.undef : vector<5xf32>
+// CHECK:           %[[VAL_30:.*]] = vector.insert_strided_slice %[[VAL_12]], %[[VAL_29]] {offsets = [0], strides = [1]} : vector<2xf32> into vector<5xf32>
+// CHECK:           %[[VAL_31:.*]] = llvm.getelementptr %[[VAL_18]][0, 7, 2] : (!llvm.ptr) -> !llvm.ptr, !llvm.struct<"class.lambda", (i16, i32, struct<"class.sycl::_V1::accessor", (ptr)>, struct<"class.sycl::_V1::vec", (array<16 x i16>)>, f32, f32, array<5 x f32>, array<5 x f32>, struct<"class.sycl::_V1::accessor", (ptr)>)>
+// CHECK:           llvm.store %[[VAL_13]], %[[VAL_31]] : vector<2xf32>, !llvm.ptr
+// CHECK:           %[[VAL_32:.*]] = vector.insert_strided_slice %[[VAL_13]], %[[VAL_30]] {offsets = [2], strides = [1]} : vector<2xf32> into vector<5xf32>
+// CHECK:           %[[VAL_33:.*]] = llvm.getelementptr %[[VAL_18]][0, 7, 4] : (!llvm.ptr) -> !llvm.ptr, !llvm.struct<"class.lambda", (i16, i32, struct<"class.sycl::_V1::accessor", (ptr)>, struct<"class.sycl::_V1::vec", (array<16 x i16>)>, f32, f32, array<5 x f32>, array<5 x f32>, struct<"class.sycl::_V1::accessor", (ptr)>)>
+// CHECK:           llvm.store %[[VAL_3]], %[[VAL_33]] : f32, !llvm.ptr
+// CHECK:           %[[VAL_34:.*]] = vector.insert %[[VAL_3]], %[[VAL_32]] [4] : f32 into vector<5xf32>
+// CHECK:           sycl.host.set_captured %[[VAL_18]][7] = %[[VAL_34]] : !llvm.ptr, vector<5xf32>
+// CHECK:           %[[VAL_35:.*]] = llvm.getelementptr %[[VAL_18]][0, 8] : (!llvm.ptr) -> !llvm.ptr, !llvm.struct<"class.lambda", (i16, i32, struct<"class.sycl::_V1::accessor", (ptr)>, struct<"class.sycl::_V1::vec", (array<16 x i16>)>, f32, f32, array<5 x f32>, array<5 x f32>, struct<"class.sycl::_V1::accessor", (ptr)>)>
+// CHECK:           %[[VAL_36:.*]] = llvm.load %[[VAL_16]] : !llvm.ptr -> !llvm.ptr
+// CHECK:           llvm.store %[[VAL_36]], %[[VAL_35]] : !llvm.ptr, !llvm.ptr
+// CHECK:           sycl.host.set_captured %[[VAL_18]][8] = %[[VAL_16]] : !llvm.ptr, !llvm.ptr (!sycl_local_accessor_base_1_21llvm2Evoid_rw)
```

Comment:
I'd rather have `local_accessor` instead of `local_accessor_base` here. Can we change `LocalAccessorBaseTypeTag` -> `LocalAccessorTypeTag` so that it returns a `LocalAccessorType` instead?

Reply:
Done. I added a TODO to the pattern stating that support is currently only for the deprecated `accessor(..., target::local)`-form.